pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -133,16 +133,50 @@ message NamespaceOption {
     optional bool host_ipc = 3;
 }
 
+// SandboxSecurityContext holds security configuration that will be applied to a sandbox.
+message SandboxSecurityContext {
+    // The configurations for the sandbox's namespaces.
+    // This will be used only if the PodSandbox uses namespace for isolation.
+    optional NamespaceOption namespace_options = 1;
+    // Optional SELinux context to be applied. Only for sandbox. Containers should
+    // confirm to SecurityContext.SelinuxOptions.
+    optional SELinuxOption selinux_options = 2;
+    // The UID to run the entrypoint of the container process.
+    // Defaults to user specified in image metadata if unspecified.
+    // Only for sandbox. Containers should confirm to SecurityContext.RunAsUser.
+    optional int64 run_as_user = 3;
+    // Indicates that the container must run as a non-root user.
+    // If true, the runtime should validate the image at runtime to ensure that it
+    // does not run as UID 0 (root) and fail to start the container if it does.
+    // If unset or false, no such validation will be performed.
+    // Only for sandbox. Containers should confirm to SecurityContext.RunAsNonRoot.
+    optional bool run_as_non_root = 4;
+    // If set, the root filesystem of the container is read-only.
+    optional bool readonly_rootfs = 5;
+    // A list of groups applied to the first process run in the sandbox, in addition
+    // to the container's primary GID.  If unspecified, no groups will be added.
+    repeated int64 supplemental_groups = 6;
+    // A special supplemental group that applies to the sandbox.
+    // Some volume types allow the Kubelet to change the ownership of that volume
+    // to be owned by the pod:
+    //
+    // 1. The owning GID will be the FSGroup
+    // 2. The setgid bit is set (new files created in the volume will be owned by FSGroup)
+    // 3. The permission bits are OR'd with rw-rw----
+    //
+    // If unset, the Kubelet will not modify the ownership and permissions of any volume.
+    optional int64 fs_group = 7;
+}
+
 // LinuxPodSandboxConfig holds platform-specific configurations for Linux
 // host platforms and Linux-based containers.
 message LinuxPodSandboxConfig {
     // The parent cgroup of the pod sandbox.
     // The cgroupfs style syntax will be used, but the container runtime can
     // convert it to systemd semantics if needed.
     optional string cgroup_parent = 1;
-    // The configurations for the sandbox's namespaces.
-    // This will be used only if the PodSandbox uses namespace for isolation.
-    optional NamespaceOption namespace_options = 2;
+    // SandboxSecurityContext holds sandbox security attributes.
+    optional SandboxSecurityContext security_context = 2;
```

Comment:
Current security context is very linux-specific, it's ok rename to `Linux{Sandbox,Container}SecurityContext`.